### PR TITLE
Script to add Uperf which is a network performance benchmark

### PR DESCRIPTION
**Pull request by hathyaga**
tool

This script will calculate the performance of the Network
adapter

Signed-off-by: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by vrbagalkote** [comment]
@hathyaga Remove wait from import statement and resend the patch.

**Comment by hathyaga** [comment]
@vrbagalkote Yup. I already knew that was causing the check to fail. I only need to re-send it again.

File: io/net/uperf_test.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        result = process.run(cmd, shell=True, ignore_status=True)
+        if result.exit_status:
+            self.fail("FAIL: Uperf Run failed")
+        for line in result.stdout.splitlines():
+            if self.peer_ip in line:
+                if 'Mb/s' in line:
+                    tput = int(line.split()[3].split('.')[0])
+                else:
+                    # Converting the throughput calculated in Gb to Mb
+                    tput = int(line.split()[3].split('.')[0]) * 1000
```
Comment (vrbagalkote) [inline io/net/uperf_test.py]: what if through put returned in some other metrics, other than Mbps and Gbps

File: io/net/uperf_test.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                                         "//github.com/uperf/uperf/"
+                                         "archive/master.zip")
+        tarball = self.fetch_asset(uperf_download, expire='7d')
+        archive.extract(tarball, self.teststmpdir)
+        self.uperf_dir = os.path.join(self.teststmpdir, "uperf-master")
+        cmd = "scp -r %s %s@%s:/tmp" % (self.uperf_dir, self.peer_user,
+                                        self.peer_ip)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.cancel("unable to copy the uperf into peer machine")
+        tmp = "cd /tmp/uperf-master;./configure ppc64le;make"
```
Comment (vrbagalkote) [inline io/net/uperf_test.py]: I think no need to use tmp here, you can use cmd only.

File: io/net/uperf_test.py (hunk 1 above)
Comment (hathyaga) [reply io/net/uperf_test.py]: Ethtool and sysfs prints the throughput in MBps. 
Considering throughput for a network card will be in MB or GB, the uperf will also calculate performance in those metrics.  I have no other metric to be considered. 

Also, I've seen a crad operating in KB speed.

File: io/net/uperf_test.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        transmitting (or receiving) data from a client. This transmit large
+        messages using multiple threads or processes.
+        """
+        if self.uperf_run == '1':
+            tmp = "cd /tmp/uperf-master/src;./uperf -s"
+            cmd = "ssh %s@%s \"%s\"" % (self.peer_user, self.peer_ip, tmp)
+            obj = process.SubProcess(cmd, verbose=False, shell=True)
+            obj.start()
+        speed = int(read_file("/sys/class/net/%s/speed" % self.iface))
+        self.expected_tp = int(self.expected_tp) * speed / 100
```
[the remark this answers is not in the document]
Comment (hathyaga) [reply io/net/uperf_test.py]: This can be considered.

File: io/net/uperf_test.py (hunk 2 above)
Comment (hathyaga) [reply io/net/uperf_test.py]: This is purely cosmetic. Rather than making the command line seem longer, I'd rather use variable "tmp".

Comment (narasimhan-v) [reply io/net/uperf_test.py]: @hathyaga What @vrbagalkote meant was, use `cmd` instead of `tmp`

File: io/net/uperf_test.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        os.chdir(self.uperf)
+        time.sleep(1)
```
Comment (narasimhan-v) [inline io/net/uperf_test.py]: Do add a comment here for the sleep

**Comment by hathyaga** [comment]
@narasimhan-v All changes suggested is taken care of. Please check.  Attached is the job log. 
[Uperf_job_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1421868/Uperf_job_log.txt)

**Comment by PraveenPenguin** [comment]
@hathyaga thanks for the patch , merging as already one ACK 
@narasimhan-v thanks for ACK

**Pull request merged**